xcm/pallet-xcm/src/lib.rs

```diff
@@ -579,8 +922,71 @@ pub mod pallet {
 				None
 			}
 		}
+
+		/// Note that a particular destination to whom we would like to send a message is unknown
+		/// and queue it for version discovery.
+		fn note_unknown_version(dest: &MultiLocation) {
+			let versioned_dest = VersionedMultiLocation::from(dest.clone());
+			VersionDiscoveryQueue::<T>::mutate(|q| {
+				if let Some(index) = q.iter().position(|i| &i.0 == &versioned_dest) {
+					// exists - just bump the count.
+					q[index].1.saturating_inc();
```

Comment:
this second value in the queue is just a priority hint?

Reply:
it's a counter for the number of times a send to the destination has been attempted.

Reply:
docs updated.